### PR TITLE
Create and delete node pool operations for gce_manager

### PR DESCRIPTION
**Pull request by mwielgus**
Ref: #225

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go [hunk 1]
```diff
@@ -294,13 +298,19 @@ func (mig *Mig) Exist() (bool, error) {
 
 // Create creates the node group on the cloud provider side.
 func (mig *Mig) Create() error {
-	return cloudprovider.ErrAlreadyExist
+	if mig.exist == false && mig.autoprovisioned == true {
```
Comment (MaciekPytel) [inline cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go]: if !mig.exists && mig.autoprovisioned

File: cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go [hunk 2]
```diff
@@ -294,13 +298,19 @@ func (mig *Mig) Exist() (bool, error) {
...
+		return mig.gceManager.createNodePool(mig.spec)
+	}
+	return fmt.Errorf("Cannot create non-autoprovisioned node group")
 }
 
 // Delete deletes the node group on the cloud provider side.
 // This will be executed only for autoprovisioned node groups, once their size drops to 0.
 func (mig *Mig) Delete() error {
-	return cloudprovider.ErrNotImplemented
+	if mig.exist == true && mig.autoprovisioned == true {
```
Comment (MaciekPytel) [inline cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go]: if mig.exist && mig.autoprovisioned

File: cluster-autoscaler/cloudprovider/gce/gce_manager.go [hunk 3]
```diff
@@ -229,6 +247,79 @@ func (m *GceManager) RegisterMig(mig *Mig) bool {
...
+// UnregisterMig unregisters mig in Gce Manager. Returns true if the node group has been removed.
+func (m *GceManager) UnregisterMig(toBeRemoved *Mig) bool {
+	m.migsMutex.Lock()
+	defer m.migsMutex.Unlock()
+
+	newMigs := make([]*migInformation, 0, len(m.migs))
+	found := false
+	for _, mig := range m.migs {
+		if mig.config.GceRef == toBeRemoved.GceRef {
+			glog.V(4).Infof("Unregistered Mig %s/%s/%s", toBeRemoved.GceRef.Project, toBeRemoved.GceRef.Zone,
```
Comment (MaciekPytel) [inline cluster-autoscaler/cloudprovider/gce/gce_manager.go]: V(0) or V(1). This should happen very rarely and it feels like a very important information.

File: cluster-autoscaler/cloudprovider/gce/gce_manager.go [hunk 4]
```diff
@@ -191,7 +205,11 @@ func (m *GceManager) fetchAllNodePools() error {
 			}
 			m.RegisterMig(mig)
 		}
-		// TODO - unregister migs
+	}
+	for _, mig := range m.getMigs() {
```
Comment (MaciekPytel) [inline cluster-autoscaler/cloudprovider/gce/gce_manager.go]: Do we have an idea how dynamically removing MIGs is going to play with clusterstate? I'm not saying we should do anything here - I would be very much against touching clusterstate from cloudprovider. Just something to think about and test.

File: cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go (hunk 1 above)
Comment (mwielgus) [reply cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go]: Done.

File: cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go (hunk 2 above)
Comment (mwielgus) [reply cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go]: Done.

File: cluster-autoscaler/cloudprovider/gce/gce_manager.go (hunk 3 above)
Comment (mwielgus) [reply cluster-autoscaler/cloudprovider/gce/gce_manager.go]: Done.

File: cluster-autoscaler/cloudprovider/gce/gce_manager.go (hunk 4 above)
Comment (mwielgus) [reply cluster-autoscaler/cloudprovider/gce/gce_manager.go]: Should be ok. In every CA iteration we are building new maps. However something may go wrong - need to check it manually once it works.

File: cluster-autoscaler/cloudprovider/gce/gce_manager.go [hunk 5]
```diff
@@ -229,6 +247,79 @@ func (m *GceManager) RegisterMig(mig *Mig) bool {
...
+	}
+	return m.fetchAllNodePools()
+}
+
+func (m *GceManager) createNodePool(spec *autoprovisioningSpec) error {
+	m.assertGKE()
+
+	// Todo: handle preemptable
+	// Todo: handle ssd
+	// Todo: handle taints
```
Comment (MaciekPytel) [inline cluster-autoscaler/cloudprovider/gce/gce_manager.go]: s/Todo/TODO 
Pretty sure when people search for TODOs they look for uppercase variant (I certainly do)

**Comment by MaciekPytel** [comment]
I'm not particularly happy about blocking for node pool creation, can you add a comment or issue to clean it up before we move NAP out of alpha?

**Pull request merged**